docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,495 @@
+> An alternative idea is to make `etcd-druid` aware of the concept of autonomous shoot clusters and give it further control into the management of static `etcd` pods.
+> The details are still subject to discussion and will become clearer in the future.
+> Since we need a "bootstrap `etcd`" anyway to get the cluster up and running, we can proceed with this GEP and its implementation in parallel while the discussion around how `etcd-druid` can take over the management is ongoing.
+
+#### `gardenadm token`
+
+After the first control plane node has been bootstrapped, it is very likely that we want to join additional nodes (either more control plane nodes to achieve high-availability, or worker nodes).
+As for regular shoot clusters, we will leverage the [`kubelet` TLS bootstrapping](https://kubernetes.io/docs/reference/access-authn-authz/kubelet-tls-bootstrapping/) process.
+This means that a [bootstrap token](https://kubernetes.io/docs/reference/access-authn-authz/bootstrap-tokens/) is used by `kubelet` to request a client certificate via the `CertificateSigningRequest` API.
+
+We will provide commands to create and manage such bootstrap tokens:
+
+- `gardenadm token generate` will generate a random bootstrap token.
+- `gardenadm token create [token]` will create a bootstrap token.
+  If no `[token]` is provided, the command will generate a random token instead.
+- `gardenadm token delete [token-id]` will delete a bootstrap token.
+- `gardenadm token list` will list all bootstrap tokens on the server.
+
+> [!NOTE]
+> `kubeadm` goes one step further and implements a discovery process for the certificate authority of the control plane.
+> For this purpose, so-called "discovery tokens" are used.
+>
+> First, the CA certificate is retrieved via an insecure connection from a `ConfigMap` that got created as part of the `kubeadm init` process.
+> Next, the downloaded CA certificate is validated based on a hash of the public key that must be provided as part of `kubeadm join`.
+> Finally, as additional validation, the CA certificate is retrieved via secure connection and compared with the CA retrieved initially.
+>
+> We might implement a similar discovery mechanism for `gardenadm` as this project evolves.
+> However, we plan to first focus on the other, "more functional" aspects.
+
+#### `gardenadm join`
+
+Joining additional worker nodes is very similar to how it works for regular shoot clusters.
+This command instruments `gardener-node-agent` reconciliation logic to write the `systemd` units and needed files for the [`gardener-node-init`](../concepts/node-agent.md#installation-and-bootstrapping) component onto the disk of the machine.
+When this unit is started, it extracts the `gardener-node-agent` binary from its container image, and creates the `systemd` unit for it.
+`gardener-node-agent` will then use the provided bootstrap token to fetch the `OperatingSystemConfig` from a `Secret` residing in the `kube-system` namespace of the cluster.
```

Comment:
Initially, the OSCs are generated and stored with help of the fake client so that the GNA's reconciliation logic (instrumented via the `gardenadm` tool) can pick them up from there and deploy the `kubelet` and static pods, etc. Once the cluster then came up, GNA must really get started on the nodes. Hence, `gardenadm init` regenerates the OSCs (result should be the same as before) and deploys them into the cluster as usual. From there, the GNA can pick up the generated `Secret` and reconcile it like for regular shoots. Joined nodes follow the same process as usual - they just connect to the API server, fetch their OSC `Secret` and apply it.